test/constraints-validation.test.ts

```diff
@@ -0,0 +1,73 @@
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '../lib';
+import { BlueprintBuilder, ControlPlaneLogTypes } from '../lib';
+import { z, ZodError } from 'zod';
+
+const longName = 'eks-blueprint-with-an-extra-kind-of-very-long-name-that-is-definitely-not-going-to-work';
+
+const app = new cdk.App();
+
+const addOns: Array<blueprints.ClusterAddOn> = [
+  new blueprints.addons.MetricsServerAddOn(),
+  new blueprints.addons.ClusterAutoScalerAddOn(),
+  new blueprints.addons.VpcCniAddOn(),
+];
+
+function getConstraintsDataSet(): [BlueprintBuilder, ZodError][] {
+  let result: [BlueprintBuilder, ZodError][] = [];
+  const blueprint1 = blueprints.EksBlueprint.builder().addOns(...addOns).id(longName);
+  
+  const blueprint1ZodError = new z.ZodError([{
```

Comment:
You can use `ZodTooBigIssue` as the data type rather that a free from object. 

```typescript
const issue : Partial<ZodTooBigIssue> = {

}
```

Reply:
After looking into this I think it is just as much code, and if not just slightly different. After adjusting my original solution to be more efficient I think its fine how it is.